app/components/search-input.js

```diff
@@ -0,0 +1,90 @@
+import Component from '@ember/component';
+
+import { later } from '@ember/runloop';
+import { denodeify } from 'rsvp';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import algoliasearch from 'algoliasearch';
+import { get, set } from '@ember/object';
+import { getOwner } from '@ember/application';
+
+const SEARCH_DEBOUNCE_PERIOD = 300;
+
+export default Component.extend({
+  classNames: ['search-input'],
+
+  page: service(),
+
+  _resultTetherConstraints: Object.freeze([
+    {
+      to: 'window',
+      pin: ['left','right']
+    }
+  ]),
+
+  init() {
+    this._super(...arguments);
+    const config = getOwner(this).resolveRegistration('config:environment');
+
+    this.client = algoliasearch(config['ember-algolia'].algoliaId, config['ember-algolia'].algoliaKey);
```

Comment:
Not too big a deal but could you destructure the variables in a separate line above? Makes it slightly nicer to read 🙂